domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1746,43 +1730,62 @@ public TotaledResults(ResultSet set, Integer total) {
     }
   }
 
-  <T> void doStreamGetQuery(SQLConnection connection, String selectQuery,
+  <T> void doStreamGetQuery(SqlConnection connection, String selectQuery,
     ResultInfo resultInfo, Class<T> clazz, List<FacetField> facets,
     Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
 
-    connection.queryStream(selectQuery, stream -> {
-      if (stream.failed()) {
-        log.error(stream.cause().getMessage(), stream.cause());
-        replyHandler.handle(Future.failedFuture(stream.cause()));
+    connection.prepare(selectQuery, ar1 -> {
+      if (ar1.failed()) {
+        log.error(ar1.cause().getMessage(), ar1.cause());
+        replyHandler.handle(Future.failedFuture(ar1.cause()));
         return;
       }
+      PreparedQuery pq = ar1.result();
+      Transaction tx = connection.begin();
+      RowStream<Row> stream = pq.createStream(50, Tuple.tuple());
       PostgresClientStreamResult<T> streamResult = new PostgresClientStreamResult(resultInfo);
-      doStreamRowResults(stream.result(), clazz, facets, resultInfo, streamResult, replyHandler);
+      doStreamRowResults(stream, clazz, facets, tx, streamResult, replyHandler);
     });
   }
 
-<T> void doStreamRowResults(SQLRowStream sqlRowStream, Class<T> clazz,
-    List<FacetField> facets, ResultInfo resultInfo,
-    PostgresClientStreamResult<T> streamResult,
-    Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
+  private List<String> populateColumnNames(Row row)
```

Comment:
The method should be static.
Can you rename the method to `columnNames` or `getColumnNames`?

Reply:
ok